packages/components/src/Autocomplete/Autocomplete.mdx

```diff
@@ -154,3 +154,58 @@ using either promises or `await` in `getOptions`.
     }
   }}
 </Playground>
+
+## Autocomplete with section headings, description and extra details
+
+You can specify an array of sections which contains options objects.
+
+<Playground>
+  {() => {
+    const options = [
+      { label: "Ships", heading: true },
```

Comment:
The array format you have isn't very descriptive of the relationship between the heading and the sublist members. E.g. we would end up having to process the array to look for headings, and we don't know easily which members of the list. We might want a more direct structure, so that we can manage the sub-lists individually. (Maybe at one point we want to hide or disable some in the list based on heading?) 

I had originally thought of the data in this format, which captures more of the membership of the headings and the sublist options, and gives us some metadata in the heading level if we need it (eg. `order`).
```
options = [
  {
    label: "Ships",
    order: 0,
    options: [
      { value, label, description, details...}, 
      { value, label, description, details...}, 
    ]
  }, {
    label: "Planets",
    order: 1,
    options: [
      { value, label, description, details...}, 
      { value, label, description, details...}, 
    ]
  }
]
```

Reply:
Just recapping our IRL discussion here for everyone else's benefit @JeffMarvin.

I went down the path above and it doesn't quite make sense to do the above yet. While the data structure looks a lot nicer the drawbacks are:
- rendering becomes an O(n^2) operation (probably not a problem with limited numbers of headings)
- arrow key traversal of the drop down is much more complex
- backwards compatibility is harder as inferring the type of data structure being passed in isn't ideal. Having to pass an empty heading for lists without headings feels weird.

We are gonna keep an eye on this and see how it's use evolves over time and potentially move to the above data structure if future needs warrant it.

Reply:
Nesting here would make this function more like the Menu component and its headers. I think it might be worth a bit more discussion here.

Reply:
We'll followup with Steve on this on Tuesday as well.